rtsp.go

```diff
@@ -138,7 +148,7 @@ func (rc *rtspCamera) closeConnection() error {
 }
 
 // reconnectClient reconnects the RTSP client to the streaming server by closing the old one and starting a new one.
-func (rc *rtspCamera) reconnectClient() (err error) {
+func (rc *rtspCamera) reconnectClient(codecInfo videoCodec) (err error) {
```

Comment:
Nit to consider:

Is passing around `codecInfo` worth omitting it from state? Will we need to access `codecInfo` in separate branches of logic?

Reply:
Either way makes sense to me.

Reply:
same. if you'd like to have less state go ahead.